File: backends/java/zappy-adapters/src/main/java/com/zappymart/adapters/graphql/GraphQlConfiguration.java
package com.zappymart.adapters.graphql;

import com.zappymart.application.accounts.IdentifyVisitor;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.core.Ordered;
import org.springframework.core.annotation.Order;
import org.springframework.graphql.execution.RuntimeWiringConfigurer;

@Configuration
@EnableConfigurationProperties(SecuritySettings.class)
public class GraphQlConfiguration {

    @Bean
    public RuntimeWiringConfigurer dateTimeScalarConfigurer() {
        return wiringBuilder -> wiringBuilder.scalar(DateTimeScalar.type());
    }

    @Bean
    @Order(Ordered.HIGHEST_PRECEDENCE)
    public OriginCheckInterceptor originCheckInterceptor(SecuritySettings securitySettings) {
        return new OriginCheckInterceptor(securitySettings);
    }

    @Bean
    @Order(Ordered.HIGHEST_PRECEDENCE + 1)
    public RequestContextInterceptor requestContextInterceptor(IdentifyVisitor identifyVisitor) {
        return new RequestContextInterceptor(identifyVisitor);
    }
}

File: backends/kotlin/zappy-host/src/main/kotlin/nl/zappymart/host/GraphQlConfiguration.kt
package nl.zappymart.host

import nl.zappymart.adapters.graphql.DateTimeScalar
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.graphql.execution.RuntimeWiringConfigurer

@Configuration
class GraphQlConfiguration {

    @Bean
    fun dateTimeScalarConfigurer() = RuntimeWiringConfigurer { wiring -> wiring.scalar(DateTimeScalar.TYPE) }
}
